### PR TITLE
Feat/implement the player and inventory class

### PR DESCRIPTION
**Pull request by mathieurobert1**
I created the Player and Inventory classes.

More Details : #25

### CODE REVIEW

**Review by AxelF44 — APPROVED** [review]

File: gui/include/Player/Inventory.hpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy GUI
+** File description:
+** Inventory
+*/
+
+#pragma once
+
+#include <iostream>
```
Comment (Thomaltarix) [inline gui/include/Player/Inventory.hpp:10]: I think it might be debug

File: gui/include/Player/Player.hpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+         *
+         * @return std::string - team name
+         */
+        std::string getTeam(void) const;
+
+        /**
+         * @brief Inventory of the player.
+         *
+         */
+        Inventory                               _inventory;
```
Comment (Thomaltarix) [inline gui/include/Player/Player.hpp:100]: Why is the inventory a public attribute ?

File: gui/src/Player/Player.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+*/
+
+#include "Player/Player.hpp"
+
+Gui::Player::Player(std::size_t id, const std::string &team, std::pair<std::size_t, std::size_t> position)
+{
+    _position = position;
+    _id = id;
+    _level = 1;
+    _team = team;
```
Comment (Thomaltarix) [inline gui/src/Player/Player.cpp:15]: I think you missed the initialization of the inventory ?

Comment (Thomaltarix) [reply gui/src/Player/Player.cpp:15]: Or maybe it's my bad and the value is by default initialized since the constructor doesn't needs parameters

**Review by Thomaltarix — COMMENTED** [review]
You can remove a little debug, and i asked a question

File: gui/include/Player/Inventory.hpp (hunk 1 above)
Comment (mathieurobert1) [reply gui/include/Player/Inventory.hpp:10]: No it's for std::size_t

File: gui/include/Player/Player.hpp (hunk 2 above)
Comment (mathieurobert1) [reply gui/include/Player/Player.hpp:100]: to not re-write all the methods of the inventory in the player

File: gui/src/Player/Player.cpp (hunk 3 above)
Comment (mathieurobert1) [reply gui/src/Player/Player.cpp:15]: By default the inventory is initialized

File: gui/include/Player/Inventory.hpp (hunk 1 above)
Comment (Thomaltarix) [reply gui/include/Player/Inventory.hpp:10]: ok nice

File: gui/src/Player/Player.cpp (hunk 3 above)
Comment (Thomaltarix) [reply gui/src/Player/Player.cpp:15]: perfect

File: gui/include/Player/Player.hpp (hunk 2 above)
Comment (Thomaltarix) [reply gui/include/Player/Player.hpp:100]: In fact, you're right

**Review by Thomaltarix — APPROVED** [review]

**Pull request merged**